src/mantine-dates/src/components/Calendar/CalendarHeader/CalendarHeader.tsx

```diff
@@ -69,6 +73,7 @@ export function CalendarHeader({
         disabled={previousMonthDisabled}
         size={iconButtonSize}
         data-mantine-stop-propagation
+        style={{ visibility: previousMonthHidden ? 'hidden' : 'visible' }}
       >
```

Comment:
I think it's unnecessary to have 2 separate props `previousMonthHidden` and `nextMonthHidden`. I would suggest making it `hiddenMonth` of type `'next' | 'prev'`.